<launch>
  <include file="$(find pr2_overhead_grasping)/launch/init_params.launch"/>
  <include file="$(find pr2_overhead_grasping)/launch/simple_grasp_pipeline.launch"/>
  <!--<include file="$(find collision_detection)/launch/sigma_detection.launch"/>-->
  <node pkg="nodelet" type="nodelet" name="coll_manager"  args="manager" output="screen">
    <param name="num_worker_threads" type="int" value="5" />
  </node>

  <node pkg="nodelet" type="nodelet" name="sensor_filter_r" args="load collision_detection/sensor_filter coll_manager" output="screen">
    <param name="arm" type="str" value="r"/>
  </node>                 
  <node pkg="nodelet" type="nodelet" name="collision_monitor_r" args="load collision_detection/collision_monitor coll_manager" output="screen">
    <param name="arm" type="str" value="r"/>
    <param name="collision_id" type="int" value="1"/>

    <param name="rf_on" type="bool" value="false"/>
    <param name="sigma_on" type="bool" value="false"/>
    <param name="error_on" type="bool" value="true"/>

    <param name="rf_thresh" type="int" value="200"/>
    <param name="rf_buf_len" type="int" value="4"/>

    <param name="sigma_buf_len" type="int" value="4"/>
    <param name="sigma_thresh" type="double" value="1.5"/>
    <param name="sigma_num" type="int" value="2"/>

    <rosparam param="min_errors"> [-0.028, -0.04, -0.04, -0.06, -0.1, -0.02, -0.025] </rosparam>
    <rosparam param="max_errors"> [0.03, 0.05, 0.06, 0.06, 0.08, 0.02, 0.025] </rosparam>

    <!--<rosparam param="classifier_topics">
        [r_arm_collision_detect,
         r_arm_gripper_direct,
         r_arm_forearm_direct,
         r_arm_uprarm_direct,
         r_arm_link_detect,
         r_arm_collision_detect_late,
         r_arm_collision_table]
    </rosparam>-->
    <rosparam param="classifier_topics"> [] </rosparam>
  </node>                 
  <node pkg="nodelet" type="nodelet" name="force_torque_monitor" args="load collision_detection/force_torque_monitor standalone_nodelet" output="screen">
    <param name="arm" type="str" value="r"/>
    <param name="z_thresh" type="double" value="4.0" />
    <param name="delay_time" type="double" value="0.15" />
    <param name="label" type="int" value="4" />
  </node>                 
  <node pkg="pr2_overhead_grasping" type="random_forest" name="$(anon random_forest_r)" output="screen">
    <param name="training_mode" type="bool" value="false"/>
    <param name="is_abs" type="bool" value="true"/>
    <param name="classify_topic" type="str" value="r_arm_features"/>
    <param name="results_topic" type="str" value="r_arm_collision_detect"/>
    <param name="loaded_topic" type="str" value="r_arm_cd_loaded"/>
    <param name="classifier_name" type="str" value="Collision Detection"/>
    <param name="classifier_id" type="int" value="0"/>
    <param name="forest_bag_name" type="str" value="collision_forest_all2.bag"/>
    <param name="bag_path" value="$(find pr2_overhead_grasping)/bags/"/>
  </node>                 
  <node pkg="pr2_overhead_grasping" type="random_forest" name="$(anon random_forest_r)" output="screen">
    <param name="training_mode" type="bool" value="false"/>
    <param name="classify_topic" type="str" value="r_arm_features"/>
    <param name="results_topic" type="str" value="r_arm_gripper_direct"/>
    <param name="loaded_topic" type="str" value="r_arm_cd_loaded"/>
    <param name="classifier_name" type="str" value="Wrist direction"/>
    <param name="classifier_id" type="int" value="1"/>
    <param name="forest_bag_name" type="str" value="collision_forest_wrist.bag"/>
    <param name="bag_path" value="$(find pr2_overhead_grasping)/bags/"/>
  </node>                 
  <node pkg="pr2_overhead_grasping" type="random_forest" name="$(anon random_forest_r)" output="screen">
    <param name="training_mode" type="bool" value="false"/>
    <param name="classify_topic" type="str" value="r_arm_features"/>
    <param name="results_topic" type="str" value="r_arm_forearm_direct"/>
    <param name="loaded_topic" type="str" value="r_arm_cd_loaded"/>
    <param name="classifier_name" type="str" value="Forearm direction"/>
    <param name="classifier_id" type="int" value="2"/>
    <param name="forest_bag_name" type="str" value="collision_forest_forearm.bag"/>
    <param name="bag_path" value="$(find pr2_overhead_grasping)/bags/"/>
  </node>                 
  <node pkg="pr2_overhead_grasping" type="random_forest" name="$(anon random_forest_r)" output="screen">
    <param name="training_mode" type="bool" value="false"/>
    <param name="classify_topic" type="str" value="r_arm_features"/>
    <param name="results_topic" type="str" value="r_arm_uprarm_direct"/>
    <param name="loaded_topic" type="str" value="r_arm_cd_loaded"/>
    <param name="classifier_name" type="str" value="Upper arm direction"/>
    <param name="classifier_id" type="int" value="3"/>
    <param name="forest_bag_name" type="str" value="collision_forest_uprarm.bag"/>
    <param name="bag_path" value="$(find pr2_overhead_grasping)/bags/"/>
  </node>                 
  <node pkg="pr2_overhead_grasping" type="random_forest" name="$(anon random_forest_r)" output="screen">
    <param name="training_mode" type="bool" value="false"/>
    <param name="classify_topic" type="str" value="r_arm_features"/>
    <param name="results_topic" type="str" value="r_arm_link_detect"/>
    <param name="loaded_topic" type="str" value="r_arm_cd_loaded"/>
    <param name="classifier_name" type="str" value="Link hit"/>
    <param name="classifier_id" type="int" value="4"/>
    <param name="forest_bag_name" type="str" value="collision_forest_link.bag"/>
    <param name="bag_path" value="$(find pr2_overhead_grasping)/bags/"/>
  </node>                 
  <node pkg="pr2_overhead_grasping" type="random_forest" name="$(anon random_forest_r)" output="screen">
    <param name="training_mode" type="bool" value="false"/>
    <param name="classify_topic" type="str" value="r_arm_features"/>
    <param name="results_topic" type="str" value="r_arm_collision_detect_late"/>
    <param name="loaded_topic" type="str" value="r_arm_cd_loaded"/>
    <param name="classifier_name" type="str" value="Collision Detection late"/>
    <param name="classifier_id" type="int" value="5"/>
    <param name="forest_bag_name" type="str" value="collision_forest_detect_late.bag"/>
    <param name="bag_path" value="$(find pr2_overhead_grasping)/bags/"/>
  </node>                 
  <node pkg="pr2_overhead_grasping" type="random_forest" name="$(anon random_forest_r)" output="screen">
    <param name="training_mode" type="bool" value="false"/>
    <param name="classify_topic" type="str" value="r_arm_features"/>
    <param name="results_topic" type="str" value="r_arm_collision_table"/>
    <param name="loaded_topic" type="str" value="r_arm_cd_loaded"/>
    <param name="classifier_name" type="str" value="Collision Detection table"/>
    <param name="classifier_id" type="int" value="6"/>
    <param name="forest_bag_name" type="str" value="collision_forest_table.bag"/>
    <param name="bag_path" value="$(find pr2_overhead_grasping)/bags/"/>
  </node>                 
</launch>
